File: src/main/scala/com/github/kmizu/yapp/runtime/TreeCharacterSet.scala
package com.github.kmizu.yapp.runtime

import java.util.TreeSet

final class TreeCharacterSet extends CharacterSet {
  def this(chars: Char*) {
    this()
    for (ch <- chars) set.add(ch)
  }

  def contains(ch: Char): Boolean = {
    return set.contains(ch)
  }

  def add(ch: Char) {
    set.add(ch)
  }

  private var set: TreeSet[Character] = new TreeSet[Character]
}

File: src/main/java/com/github/kmizu/yapp/runtime/TreeCharacterSet.java
package com.github.kmizu.yapp.runtime;

import java.util.TreeSet;

public final class TreeCharacterSet implements CharacterSet {
  private TreeSet<Character> set = new TreeSet<Character>();
  public TreeCharacterSet(char... chars) {
    for(char ch : chars) set.add(ch);
  }
  public boolean contains(char ch) {
    return set.contains(ch);
  }
  public void add(char ch) {
    set.add(ch);
  }
}
